spec/statement.dd

```diff
@@ -1220,7 +1220,7 @@ $(H2 $(LEGACY_LNAME2 SwitchStatement, switch-statement, Switch Statement))
 
 $(GRAMMAR
 $(GNAME SwitchStatement):
-    $(D switch $(LPAREN)) $(EXPRESSION) $(D $(RPAREN)) $(PSSCOPE)
+    $(D switch $(LPAREN)) $(GLINK IfCondition) $(D $(RPAREN)) $(PSSCOPE)
```

Comment:
`IfCondition` probably technically correct , in that it does the right thing, but I think the most correct thing to put is an `AssignExpression`.

Reply:
I dont think so, since `AssignExpression` dosnt covers `auto x = expr` to my knowlege, or any declaration for that matter.

Reply:
The function used in DMD, `parseAssignCondition`, is the exact same one used for `if` and `while` statements, which both use `IfCondition` in the grammar. So `IfCondition` is correct here (though perhaps it should be renamed to reflect its expanded usage).